### PR TITLE
Week1/python week1

### PR DESCRIPTION
**Pull request by mamathamereddy**

### CODE REVIEW

File: week1/solutin-to-homework.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# 4.Write a Python script to add a key to a dictionary
+# Sample Dictionary : {0: 10, 1: 20}
+# Expected Result : {0: 10, 1: 20, 2: 30}
+dict = {0: 10, 1: 20}
+dict.update({2: 30})
+print(f'updated dict:{dict}')
+
+# 5. Write a Python program to calculate body mass index.
+def calculate_BMI(height, weight):
+    return round(weight / (height * height), 2)
```
Comment (Rassibassi) [inline week1/solutin-to-homework.py]: `height * height` is right but there is a more elegant way of writing it. Look for the power to operator

File: week1/solutin-to-homework.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+# 7 .Create a tuple with different data types
+tuple_with_different_data_types= (1, 5.23, "Hello", False, ["apple", "banana", "cherry"])
+
+
+# print(tuple_with_different_data_types)
+
+# 8. Create a list of 5 city names and convert it into tuples.
+
+def convertotuple(list):
```
Comment (Rassibassi) [inline week1/solutin-to-homework.py]: here is a typo :)
a `t` is missing after `convert` :) or name it `convert2tuple` :)

HOWEVER, `tuple` is already a function, and you only created another function which wraps `tuple`, so just use `tuple` directly

File: week1/solutin-to-homework.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+a = list(set(a))
+print(f'List after Removing duplicates:{a}')
+
+
+# 10. Accept a string from user and remove the characters which have odd index values of a given string and print them.
+def remove_odd_string(str):
+    final_str = ""
+    for i in range(len(str)):
+        if i % 2 == 0:
+            final_str = final_str + str[i]
```
Comment (Rassibassi) [inline week1/solutin-to-homework.py]: `a = a + b` is the same as `a += b`

File: week1/solutin-to-homework.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+a = [10, 20, 30, 20, 10, 50, 60, 40, 80, 50, 40]
+a = list(set(a))
+print(f'List after Removing duplicates:{a}')
+
+
+# 10. Accept a string from user and remove the characters which have odd index values of a given string and print them.
+def remove_odd_string(str):
+    final_str = ""
+    for i in range(len(str)):
```
Comment (Rassibassi) [inline week1/solutin-to-homework.py]: Python can do this more elegant or for a _Pythonista_, there is a more _pythonic_ way of doing this.

In Python **a string is basically a list of characters**, or in other words, in Python like a list also a string is **iterable**,
and a list can be looped through like this:
```python
sample_list = [100,200,300]
for sample_int in sample_list:
    print(sample_int)
```
Do you see that there is no indexing `sample_list[i]` necessary anymore? You can do the same with a string

Now, you might have also seen that we lost the index number `i` within the loop. Yet, there is also a pythonic way of getting a running number into your for loop, see here:
```python
sample_list = [100,200,300]
for i, sample_int in enumerate(sample_list):
    print(i, sample_int)
```

Can you use my feedback and improve your code?

**Review by Rassibassi — COMMENTED** [review]
Good job, see my inline comments

File: week1/solutin-to-homework.py (hunk 4 above)
Comment (mamathamereddy) [reply week1/solutin-to-homework.py]: Thanks for the feedback, yes refactored my code 👍

**Pull request merged**